zapcore/json_encoder.go

```diff
@@ -145,15 +145,27 @@ func (enc *jsonEncoder) resetReflectBuf() {
 	}
 }
 
+var nullLiteralBytes = []byte("null")
+
 func (enc *jsonEncoder) AddReflected(key string, obj interface{}) error {
-	enc.resetReflectBuf()
-	err := enc.reflectEnc.Encode(obj)
-	if err != nil {
-		return err
+	// Only invoke the standard JSON encoder if there is actually something to
+	// encode; otherwise write JSON null literal directly. For now, we only
+	// apply this optimization here, but not in AppendReflected. We could apply
+	// it there as well if needed (some discussion in
```

Comment:
Since we're treating this as an optimization, I'm in favor of doin it to AppendReflected too.
It already supports `nil` entries inside slices. https://play.golang.org/p/bZb6t58qhpy

Reply:
+1